engines/scumm/imuse_digi/dimuse_engine.cpp

```diff
@@ -800,4 +800,97 @@ int IMuseDigital::diMUSESetAttribute(int attrIndex, int attrVal) {
 	return scriptParse(8, attrIndex, attrVal);
 }
 
+// Debugger utility functions
+
+void IMuseDigital::listStates() {
+	_vm->getDebugger()->debugPrintf("+---------------------------------+\n");
+	_vm->getDebugger()->debugPrintf("| stateId |         name          |\n");
+	_vm->getDebugger()->debugPrintf("+---------+-----------------------+\n");
+	if (_vm->_game.id == GID_CMI) {
+		if (_vm->_game.features & GF_DEMO) {
+			for (int i = 0; _comiDemoStateMusicTable[i].soundId != -1; i++) {
+				_vm->getDebugger()->debugPrintf("|  %4d   | %20s  |\n", _comiDemoStateMusicTable[i].soundId, _comiDemoStateMusicTable[i].name);
+			}
+		} else {
+			for (int i = 0; _comiStateMusicTable[i].soundId != -1; i++) {
+				_vm->getDebugger()->debugPrintf("|  %4d   | %20s  |\n", _comiStateMusicTable[i].soundId, _comiStateMusicTable[i].name);
+			}
+		}
+	} else if (_vm->_game.id == GID_DIG) {
+		for (int i = 0; _digStateMusicTable[i].soundId != -1; i++) {
+			_vm->getDebugger()->debugPrintf("|  %4d   | %20s  |\n", _digStateMusicTable[i].soundId, _digStateMusicTable[i].name);
+		}
+	} else if (_vm->_game.id == GID_FT) {
+		for (int i = 0; i < 48; i++) {
```

Comment:
array size maybe?

Reply:
Whoops, you're right

Reply:
I have added a 
```c++
{"ARR_END",  0,  0,    ""                    },
```
at the end of `_ftStateMusicTable` (and also changed `_ftSeqNames` accordingly), so looping through the states/sequences now looks like this:
```c++
} else if (_vm->_game.id == GID_FT) {
    for (int i = 0; strcmp(_ftStateMusicTable[i].audioName, "ARR_END"); i++) {
	  _vm->getDebugger()->debugPrintf("|  %4d   | %21s |\n", i, _ftStateMusicTable[i].name);
    }
}
```

is it ok?

Reply:
nullptr will not work?

Reply:
No, unfortunately it won't let me put a nullptr as the marker

Reply:
or add check for _ftStateMusicTable[i].name if != ""?

Reply:
Either one of these works fine:

```c++
		for (int i = 0; strcmp(_ftStateMusicTable[i].name, ""); i++) {
			_vm->getDebugger()->debugPrintf("|  %4d   | %21s |\n", i, _ftStateMusicTable[i].name);
		}
```

```c++
		for (int i = 0; _ftStateMusicTable[i].name[0]; i++) {
			_vm->getDebugger()->debugPrintf("|  %4d   | %21s |\n", i, _ftStateMusicTable[i].name);
		}
```

(by having a `	{"",         0,  0,    ""                    },` row on the state table)

Which one would you prefer?

Reply:
second one